### ISSUE TITLE
Upgrading Qt on the nightly build server

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/sqlitebrowser#1580]
The Windows and macOS nightly builds have been disabled for today, as I'm upgrading Qt on the build server for both.

On the Windows side of things, it currently has:

* Qt 5.7 (32-bit) for MSVC 2013
* Qt 5.8 (64-bit) for MSVC 2013 64-bit

Those will still be there, but Qt 5.9.6 (64-bit) and Qt 5.10.2 (64-bit) both for MSVC 2013 64-bit are being added too.

When they've finished installing (it'll prob take a few hours, as the machine is very slow), I'll update the [Win 64-bit Qt path in the CMakeLists.txt](https://github.com/sqlitebrowser/sqlitebrowser/blob/ef99ea28ada3f2cf1c2fb201602ce79aba862245/CMakeLists.txt#L38) to use Qt 5.10.2 instead.

Then I'll do similar for Qt on macOS, but likely take it to the very latest Qt release as that one should work ok.  Unlike on Windows, where I need to figure out the MSVC update too (and corresponding SQLCipher update too).

With the Qt upgrade done, our nightly builds should show the Qt-related bugs people have [previously](https://github.com/sqlitebrowser/sqlitebrowser/issues/1471#issuecomment-405059703) [mentioned](https://github.com/sqlitebrowser/sqlitebrowser/issues/1471#issuecomment-405094463), which we need to solve.

Once these bugs are solved in the nightly builds (eg `master` branch), we can pull that across to the 3.11 branch and build new binaries there for testing.

Meanwhile, I also need to get the upgrade to MSVC 2015 or 2017 sorted out (not straight forward), so we can also install the Qt (5.11.x) in order to get the improved accessibility support.

**Comment by justinclift** [comment]
The upgrade of Qt (to 5.10.1) for our Windows nightly builds has completed, with new nightly builds (19th Oct) generated using it.

The corresponding macOS upgrade of Qt (to 5.11.2) has also completed.  The nightly build script for macOS has been updated for the new Qt, and is running now.  We should have new nightly builds in ~2 hours.

**Comment by chrisjlocke** [comment]
The Win (64 bit version, anyway) runs fine, and just for confirmation, states its running Qt 5.10.1
Anything specific (not sure what accessibility bits have been added) you want me to check?

**Comment by justinclift** [comment]
For the Accessibility stuff, I think that comes with a newer (again) Qt version.  5.11 and above.  I need to upgrade the MSVC and SQLCipher versions too, before we can use Qt 5.11 (on Windows).

Hopefully I'll have time to get that done on the weekend. :smile:

**Comment by karim** [comment]
Just tested the latest nightly and I don't have any problems with it. No GUI or network error.

However, while it says Qt Version 5.10.1 in the About dialog, it is in fact is using Qt 5.8 dlls because this line need to be changed as well.

https://github.com/sqlitebrowser/sqlitebrowser/blob/4a39e343dbe32aceb7b552f27b390e2a63dd641e/installer/windows/variables.wxi#L53

**Comment by justinclift** [comment]
Oops!

Thanks @karim.  Completely missed that.  That file also has version numbers in it, which will need updating in the `master` and `v3.11.x` branches too.  Getting it done now. :smile:

**Comment by justinclift** [comment]
The macOS nightlies build server has been updated to Qt 5.11.2 now, and the binaries for today (19th Oct) generated.  Lets see how they go. :smile:

**Comment by justinclift** [comment]
The Windows nightly builds are being remade now with the fixed `variables.wxi` (just launched the script manually).  In theory, they should be up in about an hour. :smile:

**Comment by justinclift** [comment]
In theory, this upgrade is complete now.  Let's leave this issue open for an extra day, just to see if the nightlies do get automatically built correctly by themselves though. :smile:

**Comment by justinclift** [comment]
@karim @chrisjlocke If you have a minute spare, would you be ok to try the nightly builds (rebuilt) and see if any of the previous reported weirdness from Qt upgrading is still showing up?

**Comment by chrisjlocke** [comment]
Yes, the latest nightly (20 Oct) displays a 'network not accessible' error, but does continue.
However, the GUI looks odd.

![image](https://user-images.githubusercontent.com/7872218/47252768-e3866a80-d441-11e8-83e4-a3304dcc154c.png)

Rather than

![image](https://user-images.githubusercontent.com/7872218/47252780-01ec6600-d442-11e8-8ca3-bf55fd9f2d14.png)

Hovering over the DLLs, they say they're v5.10.1.  In the previous nightly, they were 5.8.

**Comment by justinclift** [comment]
Ahhh.  I wonder if there are new .dll files we should be bundling now?  Maybe there's stuff for themes and/or network support.

**Comment by chrisjlocke** [comment]
I plopped the DLLs from the 'System64' directory in the main directory just in case, but that had no effect.

**Comment by karim** [comment]
> Ahhh. I wonder if there are new .dll files we should be bundling now? Maybe there's stuff for themes and/or network support.

You are correct. After doing a quick search, I found this https://bugreports.qt.io/browse/QTBUG-65177

It looks like Qt 5.10 moved the styling to its own plugin. We have to include **qwindowsvistastyle.dll** from **styles** folder when we are building with 5.10 and up.

**Comment by karim** [comment]
I just added the required file. This should fix the GUI issue.

**Comment by justinclift** [comment]
Thanks @karim.  I've just regenerated the windows nightlies again, so should be good for testing with. :smile:

**Comment by chrisjlocke** [comment]
Thanks @justinclift .  Just tried again, and while I still get the error dialog upon starting (but can just click OK) the GUI looks better.

![image](https://user-images.githubusercontent.com/7872218/47259293-40fcd480-d49f-11e8-9314-b376b1df4fc1.png)

**Comment by karim** [comment]
Yes, I also tested it and the GUI issue is fixed. The network error is related to the auto check for new version on startup. Disabling this check stops it. Could it be something related to OpenSSL?

**Comment by SafaAlfulaij** [comment]
Hello :)
It's great that DB Browser is starting to use latest Qt versions.
May I ask to also include the Qt translations (only those that are needed, like base, network, gui, etc.)? Even if the app is translated, the standard strings aren't translated (Cut Copy Paste etc...).

Thanks for the great work!

**Comment by chrisjlocke** [comment]
> The network error is related to the auto check for new version 

Aah yes.  Thought I'd turned that off in the previous version.  The nightlies / previous versions don't display that error, and that option is turned on.  That'll be an annoying error if everyone gets that! 😉

**Comment by justinclift** [comment]
> Thought I'd turned that off in the previous version. The nightlies / previous versions don't display that error, and that option is turned on. That'll be an annoying error if everyone gets that!

Oh, I wonder if the preferences are being kept in some version dependent place in the registry?  If that's the case... that's probably a bug we'll need to fix.

**Comment by justinclift** [comment]
@SafaAlfulaij 

> May I ask to also include the Qt translations (only those that are needed, like base, network, gui, etc.)? Even if the app is translated, the standard strings aren't translated (Cut Copy Paste etc...).

Hmmm, any idea how we do that?  I was kind of thinking that should be happening already, but it's possible we're not including some files needed.  If someone can point us to the right bits to include, it should be easy to update the package building pieces to grab them. :smile:

**Comment by justinclift** [comment]
Aha, this will probably have the info we're after:

* http://doc.qt.io/qt-5/linguist-programmers.html#deploying-translations
* http://doc.qt.io/qt-5/windows-deployment.html

**Comment by justinclift** [comment]
From that "Deploying translations" link:

> In Qt 5, the .qm files were split up by module and there is a so-called meta catalog file which includes the .qm files of all modules. The name of the meta catalog file is identical to the name of Qt 4's monolithic .qm file so that existing loader code works as before provided all included .qm files are found.
>
> However, it is not necessary to always deploy all of Qt 5's .qm files. We recommend concatenating the .qm files of the modules required to one file matching the meta catalog file name using the tool lconvert in the deploy step.
> ...

**Comment by justinclift** [comment]
For the network error, I'm not sure how to go about fixing it.  My suspicion is someone with a working Windows MSVC dev environment will need to compile DB4S with debugging symbols enabled + set a breakpoint on the code around the error then step through that section one line at a time to see what's happening.

Technically that **could** be done on the Win build machine.  It's just so ancient and slow though, I tend to avoid that unless there's absolutely no other way. :wink:

**Comment by karim** [comment]
@justinclift I think the network error will be fixed when you compile DB4S with Visual Studio 2017 when using Qt 5.10 and higher.

I compiled DB4S using the latest Qt version (5.11.2) and Visual Studio 2017 and didn't get any errors. In fact the file I just added `styles/qwindowsvistastyle.dll` isn't needed anymore.

Both the GUI and network issues were fixed. I guess Visual Studio 2013 doesn't play nice with Qt 5.10.

**Comment by justinclift** [comment]
Going to close this now, as this should be all solved now.  Done as part of #1592.

@SafaAlfulaij The very latest nightly builds, generated over the last few hours, should include all of the appropriate translation files now too.  So, all of the standard strings should display in their translated form.  If that doesn't work, **please** reopen this / create a new issue / etc so it can be investigated. :smile:
